### PR TITLE
fix(optimizer): transform code incrementally

### PR DESCRIPTION
**Pull request by wmertens**
This builds on top of #6539 (which was first part of this).

Qwik should not split files before other build plugins had a chance to change the code.

The problem is that Qwik outputs "entry files" that contain multiple QRL segments, possibly from several different files, and that Rollup doesn't allow changing a file once it was loaded.

So this PR will need to enforce ordering. Not done yet.

### CODE REVIEW

**Comment by shairez** [comment]
This fixes this issue right?
#5473

File: scripts/submodule-core.ts [hunk 1]
```diff
@@ -163,9 +163,10 @@ async function submoduleCoreProd(config: BuildConfig) {
 
   console.log('🐭 core.min.mjs:', await fileSize(join(config.distQwikPkgDir, 'core.min.mjs')));
 
-  // always set the cjs version (probably imported server-side) to dev mode
   let esmCode = await readFile(join(config.distQwikPkgDir, 'core.mjs'), 'utf-8');
   let cjsCode = await readFile(join(config.distQwikPkgDir, 'core.cjs'), 'utf-8');
+  // fixup the Vite base url
+  cjsCode = cjsCode.replaceAll('undefined.BASE_URL', 'globalThis.BASE_URL||"/"');
```
Comment (PatrickJS) [inline scripts/submodule-core.ts:169]: 😅

File: packages/qwik/src/qwikloader.ts [hunk 2]
```diff
@@ -102,10 +102,12 @@ export const qwikLoader = (
       for (const qrl of attrValue.split('\n')) {
         const url = new URL(qrl, base);
         const href = url.href;
-        const symbol = url.hash[replace](/^#?([^?[|]*).*$/, '$1') || 'default';
+        const match = /^#?(([^#]+)#)?([^?[|]*).*$/.exec(url.hash);
```
Comment (PatrickJS) [inline packages/qwik/src/qwikloader.ts:105]: can we add regexp online editor link. for example https://regexr.com/ lets you also add regexp tests for example https://regexr.com/816es (don't use mine we should make a qwik account with all them saved

**Comment by PatrickJS** [comment]
this pr  https://github.com/QwikDev/qwik/pull/6017 will effect this one

**Comment by wmertens** [comment]
@PatrickJS actually I think it supercedes it - I think the problem in dev mode is chunk sources from unknown parents and this solves that. 

Although a quick fix would be to add the prefix in the existing vite-server.ts

**Comment by wmertens** [comment]
closing in favor of #6670

**Pull request closed without merging**